frontend/packages/topology/src/actions/TopologyActions.tsx

```diff
@@ -1,14 +1,19 @@
 import * as React from 'react';
-import { GraphElement } from '@patternfly/react-topology';
+import { GraphElement, Node } from '@patternfly/react-topology';
+import { referenceFor } from '@console/internal/module/k8s';
 import { ActionMenu, ActionMenuVariant, ActionServiceProvider } from '@console/shared';
+import { getResource } from '../utils';
 
 type TopologyActionsProps = {
   element: GraphElement;
 };
 
 const TopologyActions: React.FC<TopologyActionsProps> = ({ element }) => {
+  const resource = getResource(element as Node);
   return (
-    <ActionServiceProvider context={{ 'topology-actions': element }}>
+    <ActionServiceProvider
+      context={{ 'topology-actions': element, [referenceFor(resource)]: resource }}
```

Comment:
`useDeepCompareMemoize` is used to avoid unnecessary re-renders when the state is update either by the callback or by change in props. Problem with topology and `useDeepCompareMemoize` was that it was trying to stringify topology element. I've removed that in https://github.com/openshift/console/pull/9624.